### PR TITLE
AdobeSign Integration

### PR DESCRIPTION
**Pull request by Emily-ejag**
- **adding createAdobeSignAgreement function**
- **adding adobeSign cloud functions**

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [x] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [ ] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [x] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [ ] I have linked relevant issues (if any)

### CODE REVIEW

File: src/firekit.ts [hunk 1]
```diff
@@ -2443,4 +2443,45 @@ export class RoarFirekit {
...
+    } catch (error) {
+      console.error('Error creating AdobeSign agreement');
+      throw error;
+    }
+  }
+
+  async getAdobeSignAgreementStatus(agreementId: string) {
+    const cloudgetAdobeSignAgreementStatus = httpsCallable(this.admin!.functions, 'getAdobeSignAgreementStatus');
+    try {
+      const result = await cloudgetAdobeSignAgreementStatus({
```
Comment (richford) [inline src/firekit.ts:2464]:
```suggestion
      const result = await cloudGetAdobeSignAgreementStatus({
```

File: src/firekit.ts [hunk 2]
```diff
@@ -2443,4 +2443,45 @@ export class RoarFirekit {
...
+      });
+      return result;
+    } catch (error) {
+      console.error('Error creating AdobeSign agreement');
+      throw error;
+    }
+  }
+
+  async getAdobeSignAgreementStatus(agreementId: string) {
+    const cloudgetAdobeSignAgreementStatus = httpsCallable(this.admin!.functions, 'getAdobeSignAgreementStatus');
```
Comment (richford) [inline src/firekit.ts:2462]:
```suggestion
    const cloudGetAdobeSignAgreementStatus = httpsCallable(this.admin!.functions, 'getAdobeSignAgreementStatus');
```

File: src/firekit.ts [hunk 3]
```diff
@@ -2443,4 +2443,45 @@ export class RoarFirekit {
...
+    } catch (error) {
+      console.error('Error getting AdobeSign agreement status');
+      throw error;
+    }
+  }
+
+  async getAdobeSignSigningUrl(agreementId: string, email: string) {
+    const cloudgetAdobeSignSigningUrl = httpsCallable(this.admin!.functions, 'getAdobeSignSigningUrl');
+    try {
+      const result = await cloudgetAdobeSignSigningUrl({
```
Comment (richford) [inline src/firekit.ts:2477]:
```suggestion
      const result = await cloudGetAdobeSignSigningUrl({
```

File: src/firekit.ts [hunk 4]
```diff
@@ -2443,4 +2443,45 @@ export class RoarFirekit {
...
+      });
+      return result;
+    } catch (error) {
+      console.error('Error getting AdobeSign agreement status');
+      throw error;
+    }
+  }
+
+  async getAdobeSignSigningUrl(agreementId: string, email: string) {
+    const cloudgetAdobeSignSigningUrl = httpsCallable(this.admin!.functions, 'getAdobeSignSigningUrl');
```
Comment (richford) [inline src/firekit.ts:2475]:
```suggestion
    const cloudGetAdobeSignSigningUrl = httpsCallable(this.admin!.functions, 'getAdobeSignSigningUrl');
```

**Review by richford — CHANGES_REQUESTED** [review]
Looks good. Just very minor camelCase changes requested.

File: src/firekit.ts (hunk 1 above)
Comment (Emily-ejag) [reply src/firekit.ts:2464]: done!

**Review by richford — CHANGES_REQUESTED** [review]
These all look good but can I please request tsdoc docstrings for these new methods. See https://github.com/yeatmanlab/roar-firekit/blob/f8929f95324f97288a6f5f0c98af0db991726af2/src/firekit.ts#L631 for an example.

**Comment by Emily-ejag** [comment]
> These all look good but can I please request tsdoc docstrings for these new methods. See
> 
> https://github.com/yeatmanlab/roar-firekit/blob/f8929f95324f97288a6f5f0c98af0db991726af2/src/firekit.ts#L631
> 
> for an example.

Done!

**Review by richford — APPROVED** [review]

**Pull request merged**